src/nginx_manager.py

```diff
@@ -0,0 +1,260 @@
+    NginxSetupError,
+    NginxStopError,
+)
+from state import HostConfig, NginxConfig
+from utilities import execute_command
+
+logger = logging.getLogger(__name__)
+
+NGINX_SITES_ENABLED_PATH = Path("/etc/nginx/sites-enabled")
+NGINX_SITES_AVAILABLE_PATH = Path("/etc/nginx/sites-available")
+NGINX_LOG_PATH = Path("/var/log/nginx")
+
+
+# Unit test is not valuable as the module is closely coupled with nginx.
+# This should be tested with integration tests.
+
+
+def initialize() -> None:  # pragma: no cover
+    """Initialize the nginx server.
+
+    Raises:
+        NginxSetupError: Failure to set up nginx.
+    """
+    logger.info("Installing and enabling nginx")
+    # The install, systemctl enable, and systemctl start are idempotent.
+    return_code, _, stderr = execute_command(["sudo", "apt", "install", "nginx", "-yq"])
+    if return_code != 0:
+        raise NginxSetupError(f"Failed to install nginx: {stderr}")
+
+    logger.info("Clean up default configuration files")
+    _reset_sites_config_files()
+    return_code, _, stderr = execute_command(["sudo", "systemctl", "enable", "nginx"])
+    if return_code != 0:
+        raise NginxSetupError(f"Failed to enable nginx: {stderr}")
+    return_code, _, stderr = execute_command(["sudo", "systemctl", "start", "nginx"])
+    if return_code != 0:
+        raise NginxSetupError(f"Failed to start nginx: {stderr}")
+
+
+def load_config() -> None:  # pragma: no cover
+    """Load nginx configurations."""
+    if ready_check():
+        logger.info("Loading nginx configuration files")
+        # This is reload the configuration files without interrupting service.
+        execute_command(["sudo", "nginx", "-s", "reload"])
+        return
+    logger.info("Restarting nginx to load the configuration files.")
+    execute_command(["sudo", "systemctl", "restart", "nginx"])
+
+
+def stop() -> None:  # pragma: no cover
+    """Stop the nginx server.
+
+    Raises:
+        NginxStopError: Failed to stop nginx.
+    """
+    logger.info("Stopping nginx")
+    return_code, _, stderr = execute_command(["sudo", "systemctl", "stop", "nginx"])
+    if return_code != 0:
+        raise NginxStopError(f"Failed to stop nginx: {stderr}")
+
+
+def ready_check() -> bool:  # pragma: no cover
+    """Check if nginx is ready to serve requests.
+
+    Returns:
+        True if ready, else false.
+    """
+    # The return code is 0 for active and 3 for failed or inactive.
+    return_code, _, _ = execute_command(["systemctl", "status", "nginx"])
+    return return_code == 0
+
+
+def update_config(configuration: NginxConfig) -> None:
+    """Update the nginx configuration files.
+
+    Raises:
+        NginxConfigurationAggregateError: All failures related to creating nginx configuration.
+        NginxFileError: File operation errors while updating nginx configuration files.
+
+    Args:
+        configuration: The nginx locations configurations.
+    """
+    # This will reset the file permissions.
```

Comment:
Is that all the function does? Just based on the function name I was guessing it would remove customisations in the file and convert it back to a default/ blank configuration file

Reply:
The function will ensure the configuration file directories are created, empty, and has the right permissions.

I removed the code comment on line 100 as it might be unnecessary.